Jenkinsfile

```diff
@@ -1,128 +1,145 @@
 #!groovy
 
-node {
-  // System Dependent Locations
-  def mvntool = tool name: 'maven3', type: 'hudson.tasks.Maven$MavenInstallation'
-  def jdktool = tool name: 'jdk8', type: 'hudson.model.JDK'
+def jdks = ["jdk8", "jdk9"]
+def oss = ["linux"] //windows?
+def builds = [:]
+for (def os in oss) {
+  for (def jdk in jdks) {
+    builds[os+"_"+jdk] = getFullBuild( jdk, os )
+  }
+}
 
-  // Environment
-  List mvnEnv = ["PATH+MVN=${mvntool}/bin", "PATH+JDK=${jdktool}/bin", "JAVA_HOME=${jdktool}/", "MAVEN_HOME=${mvntool}"]
-  mvnEnv.add("MAVEN_OPTS=-Xms256m -Xmx1024m -Djava.awt.headless=true")
+parallel builds
```

Comment:
Is this parallel on the same JVM?
If so, this could cause problems for some tests cases that make assumptions that they have exclusive access to certain resources.

Reply:
Or parallel on the same docker image?

Reply:
it will be parallel but different nodes. so no worries.